[Fix #916] New Templating system

glogiotatidis opened the pull request:
Note to self:
 - [x] Check that default values like 'block button text' are set.
 - [x] Check that bundle keys update when images update

glogiotatidis commented on snippets/base/admin/adminmodels.py:
```diff
@@ -103,6 +103,81 @@ class AddonAdmin(admin.ModelAdmin):
+        return fields
+
+
+class SimpleTemplateInline(admin.StackedInline):
```
Not used yet, will add this inline along with some UX sauce in a following PR

metadave commented:
> ... I'm pretty lost as to what's going on and why we need new templates and whatnot

seems like a good candidate for an ADR so we can capture why this is happening

pmac commented:
I guess so. It's not exactly an architectural change though IMHO. But I would appreciate a bug or issue at least detailing what's happening and why.

metadave commented:
If it's a big enough change and not related to "architecture" (software/infra/etc), I think the DR part of ADR is still important to capture. It's probably fine as an GH issue, but we might get some more consistency by capturing as an ADR. I don't want an over-bearing processes. Up to ya'll, just my 2c.

jpetto commented:
I'm in favor of filing some sort of DR for any architecture/system/business logic change of significance. (I realize "of significance" is open to interpretation.) Personally, I find the "DR" part more important than that "A" part. 😉

glogiotatidis commented:
Valid points folks. I will include an ADR.

glogiotatidis commented:
Added ADR. Still need to work on the tests but code is there. Apologies for the huge PR folks!

glogiotatidis commented:
Oh btw I removed summernote from this PR. Created #931 to track that

glogiotatidis commented:
The migration script to convert ASRSnippets to the new system is in this gist
 
 https://gist.github.com/glogiotatidis/f143603e80e9044e0a9e67d777356147

This is a one-off, so not useful to commit in the repo. I'll get a detailed deployment plan  in #916

metadave commented:
thanks for the ADR, I appreciate the context.

pmac commented on snippets/base/admin/adminmodels.py:
```diff
@@ -103,9 +104,337 @@ class AddonAdmin(admin.ModelAdmin):
+    def snippets(self, obj):
+        """Snippets using this icon.
+
+        Needs that fancy code b/c icons have multiple relations with multiple Templates.
```
Fancy! 🎩

metadave replied:
fancy, indeed

glogiotatidis replied:
Moved the fancy code in `Icon` model to make it re-usable. ;)

glogiotatidis commented:
Updated the [migration script gist](https://gist.github.com/glogiotatidis/f143603e80e9044e0a9e67d777356147) to set default values to migrated snippets

glogiotatidis commented:
The template are based on the ASR schemas,e.g.
e.g.
https://github.com/mozilla/activity-stream/blob/master/content-src/asrouter/templates/SimpleSnippet/SimpleSnippet.schema.json

glogiotatidis commented:
OK this is done!

Added more tests, removed unused things and manually tested using a database dump from the real db.

@pmac final review at your time please! :)

glogiotatidis commented on snippets/urls.py:
```diff
@@ -43,7 +43,8 @@ def robots_txt(request):
         response = static_serve(*args, **kwargs)
         response['Access-Control-Allow-Origin'] = '*'
-        if settings.BUNDLE_BROTLI_COMPRESS:
+        if ((settings.BUNDLE_BROTLI_COMPRESS and
```
Intentional. One way to work around [E129](https://lintlyci.github.io/Flake8Rules/rules/E129.html) which works with my editor

glogiotatidis commented:
> This all looks good to me Giorgos. It's a lot, and complex, so I could well have missed something, especially with real model inheritance... that's always tricky. It's a lot of good work though! As far as I'm concerned it's ready to give a good try +1

Agreed, I know it's complex. I've a detailed migration plan in #916 to make sure that I test all the things and I've communicated with the  LCM team that this is coming.

Thanks for your time and review!
